hack/run_test

```diff
@@ -377,43 +450,30 @@ function run_master_restart_test() {
 
   echo "Testing failed master restart"
   local cid_suffix="mrestart"
-  local network=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
-  local docker_args="--net=$network"
-  local cluster_args="$docker_args -e POSTGRESQL_ADMIN_PASSWORD=pass -e POSTGRESQL_MASTER_USER=$PGUSER -e POSTGRESQL_MASTER_PASSWORD=$PASS"
+  local cluster_args="-e POSTGRESQL_ADMIN_PASSWORD=pass -e POSTGRESQL_MASTER_USER=$PGUSER -e POSTGRESQL_MASTER_PASSWORD=$PASS"
   local table_name="t1"
-  local master_ip=
-  local slave_cids=
 
-  local volume_dir=$(create_volume)
+  local volume_dir
+  create_pg_volume_dir
   local master_args="-v ${volume_dir}:/var/lib/pgsql/data:Z"
 
-  DOCKER_NETS="${DOCKER_NETS:-} $network"
-
-  # Setup the cluster
+  # Setup the cluster, keep exported vars local only.
+  local network network_ip master_ip slave_ips= psql_docker_args
   slave_num=2 setup_replication_cluster
 
   # Check if the master knows about the slaves
   CONTAINER_IP=$master_ip
   test_slave_visibility
 
-  echo "Kill the master and create a new one"
-  local cidfile=$CIDFILE_DIR/master-$cid_suffix.cid
-  docker kill $(cat $cidfile)
-  # Don't forget to remove its .cid file
-  rm $cidfile
-
-  run_master $cid_suffix
-  CONTAINER_IP=$(get_container_ip master-$cid_suffix.cid)
-  echo "New master is: $CONTAINER_IP"
+  : "Kill the master and create a new one"
+  docker kill "$(cat "$CIDFILE_DIR/master-$cid_suffix.cid")"
```

Comment:
While doing some testing of running this test case in parallel, sometimes the test would fail with an "IP already in use" error. We should add a `sleep 1` over here to wait a while for docker to properly clean up after the killed container before trying to run it again with the same IP.